### PR TITLE
Ubuntu vivid should use systemd not upstart

### PR DESCRIPTION
**Pull request by gabriel403**

### CODE REVIEW

File: manifests/params.pp [hunk 1]
```diff
@@ -351,7 +351,14 @@
 
   case $::operatingsystem {
     'Ubuntu': {
-      $server_service_provider = upstart
```
Comment (igalic) [inline manifests/params.pp]: server is the only thing that runs a service, so it's kinda redundant.

File: manifests/params.pp [hunk 2]
```diff
@@ -351,7 +351,14 @@
...
+      case $::lsbdistcodename {
+        'vivid': {
+          $server_service_provider = systemd
+        }
+        default: {
+          $server_service_provider = upstart
+        }
+      }
```
Comment (igalic) [inline manifests/params.pp]: and this whole thing can be replaced with a selector:
```puppet
$server_service_provider = $::lsbdistcodename {
     'vivid' => 'systemd',
     default: 'upstart'
}
```
and now the problem becomes extending it for future versions…

i'm really starting to feel the burden of the systemd breakage in debuntu… :(

Comment (gabriel403) [inline manifests/params.pp]: @igalic thanks for the tip! Will implement

**Comment by DavidS** [comment]
Please don't use codenames for these checks, but a open range on the release version. We can expect that from now on all Ubuntu releases will use systemd.

**Comment by gabriel403** [comment]
@DavidS would be glad to, but you're gonna have to point me in the right direction for that

**Comment by gabriel403** [comment]
@DavidS `versioncmp($::lsbdistrelease, '14.10') == 1` ?

**Comment by DavidS** [comment]
Almost :-) `lsbdistrelease` is not available always, but `operatingsystemmajrelease` is. to aid readability use `> 0 `:

```
versioncmp($::operatingsystemmajrelease, '14.10') > 0
```

**Comment by bmjen** [comment]
Hi @gabriel403, any progress on this PR?

**Comment by gabriel403** [comment]
@bmjen hi sorry! Will sort this tonight :)

**Comment by gabriel403** [comment]
@bmjen @DavidS sorry about the huge delay on this, should be sorted now, if you can validate it's as you expect I'll rebase and squash

**Comment by DavidS** [comment]
@gabriel403 looks good, except for the missing `{` :

```
Syntax error at 'server_service_provider'; expected '}' at /home/travis/build/puppetlabs/puppetlabs-mysql/manifests/params.pp:363
```

**Comment by gabriel403** [comment]
@DavidS looks like we have a winner? Anything else you'd like me to do?

**Comment by DavidS** [comment]
Celebrate? :-)  Thanks!

**Comment by gabriel403** [comment]
Woo! Thanks @DavidS

**Pull request merged**